core/irCheck.ml

```diff
@@ -547,14 +593,19 @@ struct
               end
             else
               begin
-              ensure (eq_types (`Value orig) (o#extract_type_equality_context ()) (vt, t) || is_sub_type (vt, t)) (Printf.sprintf "coercion error: %s is not a subtype of %s"
-                                         (string_of_datatype vt) (string_of_datatype t)) (`Value orig);
+              ensure (eq_types (SVal orig) (o#extract_type_equality_context ()) (vt, t) || is_sub_type (vt, t)) (Printf.sprintf "coercion error: %s is not a subtype of %s"
+                                         (string_of_datatype vt) (string_of_datatype t)) (SVal orig);
               Coerce (v, t), t, o
               end
 
```

Comment:
Why an empty line here?

Reply:
Trying to separate the inner function visually from the other one